openapi/src/main/java/io/helidon/openapi/ExpandedTypeDescription.java

```diff
@@ -0,0 +1,303 @@
+        result.setPropertyUtils(PROPERTY_UTILS);
+        return result;
+    }
+
+    private ExpandedTypeDescription(Class<? extends Object> clazz, Class<?> impl) {
+        super(clazz, null, impl);
+        this.impl = impl;
+    }
+
+    /**
+     * Adds property handling for a {@code $ref} reference.
+     *
+     * @return this type description
+     */
+    ExpandedTypeDescription addRef() {
+        PropertySubstitute sub = new PropertySubstitute("$ref", String.class, "getRef", "setRef");
+        sub.setTargetType(impl);
+        substituteProperty(sub);
+        return this;
+    }
+
+    /**
+     * Adds property handling for extensions.
+     *
+     * @return this type description
+     */
+    ExpandedTypeDescription addExtensions() {
+        PropertySubstitute sub = new PropertySubstitute("extensions", Map.class, "getExtensions", "setExtensions");
+        sub.setTargetType(impl);
+        substituteProperty(sub);
+        return this;
+    }
+
+    @Override
+    public Property getProperty(String name) {
+        return isExtension(name) ? new ExtensionProperty(name) : super.getProperty(name);
+    }
+
+    Property getPropertyNoEx(String name) {
+        try {
+            Property p = getProperty("defaultValue");
+            return p;
+        } catch (YAMLException ex) {
+            if (ex.getMessage().startsWith("Unable to find property")) {
+                return null;
+            }
+            throw ex;
+        }
+    }
+
+    @Override
+    public Object newInstance(String propertyName, Node node) {
+        Property p = getProperty(propertyName);
+        if (p.getType().isEnum()) {
+            @SuppressWarnings("unchecked")
+            Class<Enum> eClass = (Class<Enum>) p.getType();
+            String valueText = ScalarNode.class.cast(node).getValue();
+            for (Enum e : eClass.getEnumConstants()) {
+                if (e.toString().equals(valueText)) {
+                    return e;
+                }
+            }
+        }
+        return super.newInstance(propertyName, node);
+    }
+
+    @Override
+    public boolean setupPropertyType(String key, Node valueNode) {
+        return setupExtensionType(key, valueNode) || super.setupPropertyType(key, valueNode);
+    }
+
+    void addExcludes(String... propNames) {
+        if (excludes == Collections.<String>emptySet()) {
+            excludes = new HashSet<String>();
+        }
+        for (String propName : propNames) {
+            excludes.add(propName);
+        }
+    }
+
+    /**
+     * Returns the implementation class associated with this type descr.
+     *
+     * @return implementation class
+     */
+    Class<?> impl() {
+        return impl;
+    }
+
+    boolean hasDefaultProperty() {
+        return getPropertyNoEx("defaultValue") != null;
+    }
+
+    private boolean setupExtensionType(String key, Node valueNode) {
+        if (isExtension(key)) {
+            switch (valueNode.getNodeId()) {
+                case sequence:
```

Comment:
These `case` values surprised me; where are they coming from?  Obviously everything compiles so it's fine but maybe a quick one-liner comment somewhere explaining their provenance.

Reply:
Comment added.